README.md

```diff
@@ -1 +1,160 @@
+
+    A 4 element immutable array with fields `theta`, `x`, `y`, and `z` to store the rotation angle and axis of the rotation.  This is a typealias of a [FixedSizeArrays](https://github.com/SimonDanisch/FixedSizeArrays.jl) `FixedVectorNoTuple{4, T}`.
+
+
+
+3. **EulerAngles** `EulerAngles{Order <: TaitByranOrder, T <: AbstractFloat}`
+
+    A 3 element immutable array which stores the Euler angles with [**Tait Byran**](https://en.wikipedia.org/wiki/Euler_angles#Tait.E2.80.93Bryan_angles) angle ordering (e.g. `EulerXYZ`) encoded by the template parameter `Order`.   This is a typealias of a [FixedSizeArrays](https://github.com/SimonDanisch/FixedSizeArrays.jl) `FixedVectorNoTuple{3, T}`.
+
+
+
+4. **ProperEulerAngles** `ProperEulerAngles{Order <: ProperEulerOrder, T <: AbstractFloat}`
+
+    A 3 element immutable array which stores the Euler angles with [**Proper Euler**](https://en.wikipedia.org/wiki/Euler_angles#Conventions) angle ordering (e.g. `EulerXYX`) encoded by the template paramter `Order`.   This is a typealias of a [FixedSizeArrays](https://github.com/SimonDanisch/FixedSizeArrays.jl) `FixedVectorNoTuple{3, T}`.
+
+
+
+5. **Quaternions** `Quaternion{T <: Real}`
+
+    A 4 element immutable array containing the Quaternion representation of the rotation.  This uses the [Quaternions](https://github.com/JuliaGeometry/Quaternions.jl) package.
+
+
+
+6. **Stereographic Quaternion Projection** `SpQuat{T <: AbstractFloat}`
+
+    A 3 element immutable array containing the stereographic projection of a unit Quaternion.  This gives a compact three element representation of the rotation, but unlike EulerAngles the derivitives of the rotation matrix w.r.t. the `SpQuat` parameters are rational functions.  This makes the `SpQuat` type a good choice for use in optimization problems.
+
+    This projection can be visualized as a pin hole camera, with the pin hole matching the Quaternion `[-1,0,0,0]` and the image plane containing the origin and having normal direction `[1,0,0,0]`.  The "no rotation" `Quaternion(1.0,0,0,0)` then maps to the `SpQuat(0,0,0)`
+
+
+
+
+### Example Usage
+
+```julia
+
+    # create a rotation matrix
+    R = RotMatrix(eye(3))
+
+    # create a point
+    using FixedSizeArrays
+    X = Vec{3, Float64}(1,2,3)
+
+    # convert to Euler Angles (using the default ordering scheme)
+    ea = EulerAngles(R)
+    Xo = ea * X
+
+    # convert to Euler Angles specifying the angle order
+    ea = EulerAngles{Rotations.EulerXYZ}(R)
+    Xo = ea * X
+
+    # convert to proper Euler Angles (using the default ordering scheme)
+    pea = ProperEulerAngles(R)
+    Xo = pea * X
+
+    # convert to proper Euler Angles specifying the angle order
+    ea = ProperEulerAngles{Rotations.EulerXYX}(R)
+    Xo = ea * X
+
+    # a Quaternion
+    q = Quaternion(R)
+    Xo = q * X
+
+    # Stereo graphic projection of a quaternion (recommended for optimization problems)
+    spq = SpQuat(R)
+    Xo = spq * X
```

Comment:
When we had the discussion I'd forgotten that the template for * is for
Vec{3, Real}, so cases like:
``` q = Quaternion(1.0, 0.0, 0.0, 0.0)
qv = Vec(q,q,q)
@which(q * qv)
*{T<:Number}(x::T<:Number,
y::FixedSizeArrays.FixedArray{T<:Number,NDim,SIZE}) at
/home/andrew/.julia/v0.4/FixedSizeArrays/src/ops.jl:76
```
doesn't get handled by this package.  I haven't managed to think of an
example where it would cause unexpected results with Reals.  Also I like
the shorter syntax of *

What do you guys think? @AndyFerris





On 9 May 2016 at 14:18, Chris Foster <notifications@github.com> wrote:

> In README.md
> <https://github.com/FugroRoames/Rotations.jl/pull/2#discussion_r62449113>:
>
> > +
> > +    # convert to proper Euler Angles (using the default ordering scheme)
> > +    pea = ProperEulerAngles(R)
> > +    Xo = pea * X
> > +
> > +    # convert to proper Euler Angles specifying the angle order
> > +    ea = ProperEulerAngles{Rotations.EulerXYX}(R)
> > +    Xo = ea * X
> > +
> > +    # a Quaternion
> > +    q = Quaternion(R)
> > +    Xo = q * X
> > +
> > +    # Stereo graphic projection of a quaternion (recommended for optimization problems)
> > +    spq = SpQuat(R)
> > +    Xo = spq * X
>
> I can't remember - did we decide to ditch * in favour of rotate() ?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/FugroRoames/Rotations.jl/pull/2/files/fb3a7882257b93afed97cfaa9bf56669bbf410dc#r62449113>
>